File: src/co/edu/uptc/music/logic/models/Artist.java
package co.edu.uptc.music.logic.models;

public class Artist extends User {

    private GenreType genre;
    private String img;

    public Artist(String id, UserType type, String name, String email, String username, String
            password, GenreType genre, String img) {
        super(id, type, name, email, username, password);
        this.genre = genre;
        this.img = img;
    }

    public GenreType getGenre() {
        return genre;
    }

    public String getImg() {
        return img;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Artist)) return false;
        Artist artist = (Artist) o;
        return getName().equals(artist.getName());
    }

}

File: src/co/soundwave/models/Artist.kt
package co.soundwave.models

data class Artist(
    val id: Int,
    val name: String,
    val nick: String,
    val email: String,
    val password: String
                 ) {
    
    fun validate(otherPassword: String): Boolean =
        password.equals(otherPassword, true)
}